frontend/src/concepts/distributedWorkloads/DistributedWorkloadsContext.tsx

```diff
@@ -109,6 +111,7 @@ export const DistributedWorkloadsContextProvider =
           projectCurrentMetrics,
           refreshAllData,
           namespace,
+          projectDisplayName: project ? getProjectDisplayName(project) : namespace,
```

Comment:
Namespace and Project are synonyms as far as the UI is concerned. This is very weird logic.

If you cannot find a project... how do you have a namespace? You shouldn't render your context if you don't have the bare min.  

So an error if you're without the min required for anything to work.

I get this is not a change you're doing with this PR -- but this exposed a gap in your code you need to close with this PR or do another PR and close it there and put this on top. This blocks this from merging.

Reply:
I think the reason for the fallback here was strictly a TypeScript one: `project` can be null because it comes from a `projects.find()` and TS doesn't know that we will definitely always find a project. I could use a `!` here to assert it's always defined, but then what if there is some error loading projects from the API and we really do have a null value? suddenly we have a null value at run time passed down for a statically non-null type. .... although thinking about it, if we have no projects this whole page won't render anyway, so it's moot. Maybe a `!` assertion up where `project` is defined on line 53 is the way to go?
```ts
const project = projects.find(byName(namespace))!;
```